gui/src/hooks/useDialog.ts

```diff
@@ -0,0 +1,22 @@
+import { invoke } from "@tauri-apps/api/tauri";
+import { useCallback } from "react";
+
+import { useInvoke } from "./tauri";
+
+export function useDialog<T>(id: number) {
+  const { data } = useInvoke<T>("dialog_get_payload", { id });
+
+  const accept = useCallback(
+    (payload: unknown) =>
+      invoke("dialog_finish", { id, result: { Ok: payload } }),
+    [id]
+  );
+
+  const reject = useCallback(
```

Comment:
Can you remove the `useCallback`? These really just for optimizations.

Reply:
I was researching this (after noticing you didn't use them).
one of the rules-of-thumb I read about this was: use `useCallback` if you pass the callbacks down to a child

is that not applicable here? or rather, is there any downside to using it here? feels like it could eventually become one of those hard to track bugs

Reply:
I didn't find the talk, but here's a [blog post about it](https://kentcdodds.com/blog/usememo-and-usecallback).

Reply:
fixed